lib/features/mailbox/domain/usecases/delete_multiple_mailbox_interactor.dart

```diff
@@ -24,6 +31,70 @@ class DeleteMultipleMailboxInteractor {
 
       final currentMailboxState = await _mailboxRepository.getMailboxState(session, accountId);
 
+      final mailboxResponses = await _mailboxRepository.getAllMailbox(session, accountId).toList();
+
+      final listUnsubscribedMailbox = mailboxResponses.expand((mailboxResponse) {
+        final presentationMailboxes = mailboxResponse.mailboxes
+          ?.map((mailbox) => mailbox.toPresentationMailbox()).toList()
+          ?? List<PresentationMailbox>.empty();
+        return presentationMailboxes.listUnsubscribedMailboxes;
+      }).toSet();
+
+      if (listUnsubscribedMailbox.isNotEmpty) {
+        Map<MailboxId, MailboxNode> mailboxDictionary = HashMap();
+        final unsubscribedTree = MailboxTree(MailboxNode.root());
+
+        for (var mailbox in listUnsubscribedMailbox) {
+          if (mailbox.parentId != null) {
+            mailboxDictionary[mailbox.id] = MailboxNode(mailbox);
+          }
+        }
+
+        for (var mailbox in listUnsubscribedMailbox) {
+          final parentId = mailbox.parentId;
+          if (parentId != null) {
+            final parentNode = mailboxDictionary[parentId];
+            final node = mailboxDictionary[mailbox.id];
+            if (node != null) {
+              if (parentNode != null) {
+                parentNode.addChildNode(node);
+              } else {
+                mailboxDictionary[parentId] = MailboxNode(PresentationMailbox(parentId));
+                unsubscribedTree.root.addChildNode(mailboxDictionary[parentId]!);
+                mailboxDictionary[parentId]!.addChildNode(node);
+              }
+            }
+          }
+        }
+
+        List<MailboxId> visitedUnsubscribedMailboxIds = [];
+
+        for (var mailboxId in listMailboxIdToDelete) {
+          if (visitedUnsubscribedMailboxIds.contains(mailboxId)) {
+            continue;
+          } else {
+            final matchedNode = unsubscribedTree.findNode((node) => node.item.id == mailboxId);
+
+            if (matchedNode != null) {
+              final descendantIds = matchedNode.descendantsAsList().mailboxIds;
+              final descendantIdsReversed = descendantIds.reversed.toList();
+              descendantIdsReversed.removeLast();
+
+              if (mapMailboxIdToDelete.containsKey(mailboxId)) {
+                mapMailboxIdToDelete[mailboxId]!.insertAll(mapMailboxIdToDelete[mailboxId]!.length - 1, descendantIdsReversed);
+              } else {
+                for (var listIdToDelete in mapMailboxIdToDelete.values) {
+                  if (listIdToDelete.contains(mailboxId)) {
+                    listIdToDelete.insertAll(listIdToDelete.indexOf(mailboxId), descendantIdsReversed);
+                  }
+                }
+              }
+              visitedUnsubscribedMailboxIds.addAll(descendantIdsReversed);
+            }
+          }
+        }
+      }
```

Comment:
We should separate the function to make it easier to understand. If possible, write this unitest for interactor